lib/block.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const stream = require('stream');
+const StringifierStream = require('./streams/stringifier-stream');
+const Fragment = require('./fragment');
+const FRAGMENT_EVENTS = ['start', 'response', 'end', 'error', 'timeout', 'fallback', 'warn'];
+
+module.exports.buildBlock = function buildBlock(request, originalContext) {
+    // NOTE: should there be a separation between context as something global
+    // vs. context as quilt/tailor-specific term?
+    let context = originalContext;
+
+    const {
+        maxAssetLinks,
+        asyncStream,
+        handleTag,
+        requestFragment,
+
+        // Should the two below be one???
```

Comment:
can you explain what you mean here? having a single option instead of 3 separate options?

Reply:
Yeah, it's a note to myself: we have many pipe-related options, I was wondering if some of the options can be joined (the function part with the attribute part, cause we just pass one to another :D )